File: Code blocks/t_primes.c
#include<stdio.h>
#include<math.h>
int main(){
    long long int t,i,j,k,n,c;
    scanf("%lld",&t);
    while(t){
        scanf("%lld",&n);
        if(n<=10){
            if(n==4||n==9){
                printf("YES");
            }
            else{
                printf("NO");
            }
        }
        else{
            c=sqrt(n);
            int z=0;
            if(c*c==n){
                for(i=2;i<(int)(ceil(sqrt(c)))+1;i++){
                    if(c%i==0){
                        z=1;
                        break;
                    }
                    else{
                        continue;
                    }
                }
                if(z==1){
                    printf("NO");

                }
                else{
                    printf("YES");
                }
            }
            else{
                printf("NO");
            }
        }
        printf("\n");
        t--;
    }
}

File: Python/t_primes.py
import math
def checker(n):
    z=0
    for i in range(2,int(math.ceil(math.sqrt(n)))+1):
        if n%i==0:
            z=1
            break
        else:
            continue
    return z

t=int(input())
a=list(map(int,input().split()))
for i in range(0,t):
    c=int(math.sqrt(a[i]))
    if c==math.sqrt(a[i]):
        z=checker(c)
        if z==0:
            print("YES")
        else:
            print("NO")
    else:
        print("NO")
